units/migrate_volume_job.go

```diff
@@ -0,0 +1,298 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/cloud"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const (
+	volumeMigrationJobName = "volume-migration"
+)
+
+func init() {
+	registry.AddJobType(volumeMigrationJobName,
+		func() amboy.Job { return makeVolumeMigrationJob() })
+}
+
+type volumeMigrationJob struct {
+	job.Base      `bson:"job_base" json:"job_base" yaml:"job_base"`
+	VolumeID      string             `bson:"volume_id" yaml:"volume_id"`
+	ModifyOptions cloud.SpawnOptions `bson:"modify_options" json:"modify_options" yaml:"modify_options"`
+
+	VolumeDetached        bool   `bson:"volume_detached"`
```

Comment:
Since these aren't stored with the job do they need to be public? (I might be misunderstanding what's going on here though)

Reply:
To the best of my understanding, we need to make these structs public so that they are preserved on subsequent attempts of the job (this is a characteristic of using a remote queue). See "A Note on Idempotency" in the [Amboy doc](https://docs.google.com/document/d/1FJ1cj60H6gQ3Cw0M-Wx149YSvRTXbWgSLZePhpiVlII/edit?usp=sharing). If the `VolumeDetached` field isn't preserved in the job document, we won't be able to pick up the job in its current state.